test/cpp-tableau-loader/src/protoconf/hub.pc.h

```diff
@@ -35,16 +35,20 @@ struct HubOptions {
 
 class Hub {
  public:
-  Hub(const HubOptions* options = nullptr)
-      : msger_container_(std::make_shared<MessagerContainer>()), options_(options ? *options : HubOptions{}) {}
+  Hub(std::shared_ptr<const HubOptions> options = nullptr);
+
+  // Init resets the hub's options.
+  void Init(MessagerContainerProvider provider, Filter filter = nullptr);
```

Comment:
If Init resets the hub's options, then the parameter should be `std::shared_ptr<const HubOptions>` for scalable future extensions and consistent understanding.